### ISSUE TITLE
shortcut for Iterable<E: Comparable<E>>.max()

### ISSUE THREAD

**Issue by robstoll** [robstoll/atrium#162]
*Platform* (jvm, jdk8, js, android): all

## Code related feature
```
expect(listOf(1,2,3)).max().toBe(3) 
expect(listOf(1,2,3)).max {
   isGreaterThan(3)  
   isLessThan(5)
}

//instead of

expect(listOf(1,2,3)).feature { f(it::max) }.notToBeNull().toBe(3)
expect(listOf(1,2,3)).feature({ f(it::max) }) {
  notToBeNull {
    isGreaterThan(3)  
    isLessThan(5)
  }
}
```
However, in reporting we want to see the following in case there aren't any elements in the iterable:
```
expect: List()
* max: !! cannot be determined, empty Iterable
  >> to be: 3
```

Following the things you need to do:

*domain*
- extend IterableAssertions with a function `max` (see CollectionAssertions.size as a guideline)
- modify IterableAssertionsBuilder, delegate to iterableAssertions (see other functions as a guideline)
- delegate implementation to robstoll-lib in IterableAssertionsImpl (see other functions as a guideline)

*lib*
- implement _max in iterableAssertions by using `ExpectImpl.feature.extractor` (see listAssertions -> _get)

*api*  (only fluent-en_GB)
- provide a val which returns Expect<E> (see collectionAssertions.kt as a guideline)
- provide a fun which expects an `assertionCreator`-lambda and returns Expect<T> where `T : Iterable<E>, E : Comparable<E>` (see collectionAssertions.kt as a guideline)
- augment IterableFeatureAssertionsSpec with `max` in specs-common (see for instance CollectionFeatureAssertionsSpec) and extend it in atrium-api-fluent-en_GB-common/src/test

## Your first contribution?
- Write a comment `I'll work on this` if you would like to take this issue over. 
  This way we get the chance to revise the description in case things have changed in the meantime,
  we might give you additional hints and we can assign the task to you, so that others do not start as well.
- See [Your first code contribution](https://github.com/robstoll/atrium/blob/master/.github/CONTRIBUTING.md#your-first-code-contribution) for guidelines.  
- Do not hesitate to ask questions here or to contact us via [Atrium's slack channel](https://kotlinlang.slack.com/team/U3DE1TXKP) if you need help
  ([Invite yourself](https://slack.kotlinlang.org/) in case you do not have an account yet).

**Comment by sebastienD** [comment]
I'll work on this

**Comment by robstoll** [comment]
@sebastienD do you need help?

**Comment by Megamiun** [comment]
Which should be the behaviour in the case of an empty list, when there is no max?

**Comment by robstoll** [comment]
@Megamiun good point. I have not thought about that. In this case it should fail as well with an assertion explaining the reason; something like:
```
expect: List()
* max: !! cannot be determined, empty Iterable
  >> to be: 3
```

**Comment by robstoll** [comment]
Implemented by fabde3df968e74a6128e392aa64c1967e4d1565f
